### PR TITLE
Allow proxy agent to connect to non local server.

### PR DESCRIPTION
**Pull request by cheftako**
Added configuration for where the proxy server is.
Also improved cert related errors for easier debug.
Added comments to sample cert generation in make script to make it a little clearer.

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -55,47 +55,59 @@ func main() {
...
+	proxyServerHost string
+	proxyServerPort int
 }
 
 func (o *GrpcProxyAgentOptions) Flags() *pflag.FlagSet {
 	flags := pflag.NewFlagSet("proxy-agent", pflag.ContinueOnError)
 	flags.StringVar(&o.agentCert, "agentCert", o.agentCert, "If non-empty secure communication with this cert.")
 	flags.StringVar(&o.agentKey, "agentKey", o.agentKey, "If non-empty secure communication with this key.")
 	flags.StringVar(&o.caCert, "caCert", o.caCert, "If non-empty the CAs we use to validate clients.")
+	flags.StringVar(&o.proxyServerHost, "proxyServerHost", o.proxyServerHost, "The hostname to use to connect to the proxy-server.")
```
Comment (andrewsykim) [inline cmd/agent/main.go]: Should this be:
```
flags.StringVar(&o.proxyServerHost, "proxyServerHost", proxyServerHost, "The hostname to use to connect to the proxy-server.")
```

(default set to local host)

File: cmd/agent/main.go [hunk 2]
```diff
@@ -55,47 +55,59 @@ func main() {
 }
 
 type GrpcProxyAgentOptions struct {
+	// Configuration for authenticating with the proxy-server
 	agentCert string
 	agentKey  string
 	caCert    string
+
+	// Configuration for connecting to the proxy-server
+	proxyServerHost string
```
Comment (andrewsykim) [inline cmd/agent/main.go]: rename to `defaultProxyServerHost` to avoid confusion in above comment?

File: cmd/agent/main.go (hunk 1 above)
Comment (cheftako) [reply cmd/agent/main.go]: good catch

**Pull request merged**